### PR TITLE
Make TestEndpoint GSO compatible

### PR DESCRIPTION
**Pull request by Matthias247**
The TestEndpoint so far tried to send a single big datagram instead of using
proper GSO sending to emit multiple datagrams.

In order to avoid relying on platform GSO support, this change will simply
split up GSO transmissions into multiple Transmits in the endpoint.

### CODE REVIEW

File: quinn-proto/src/tests/util.rs [hunk 1]
```diff
@@ -385,6 +389,32 @@ pub fn min_opt<T: Ord>(x: Option<T>, y: Option<T>) -> Option<T> {
...
+        let contents = transmit.contents[offset..end].to_vec();
+        transmits.push(Transmit {
+            destination: transmit.destination,
+            ecn: transmit.ecn,
+            contents,
+            segment_size: None,
+            src_ip: transmit.src_ip,
+        });
+
+        offset += segment_size;
```
Comment (Ralith) [inline quinn-proto/src/tests/util.rs:412]: nit:
```suggestion
        offset = end;
```

File: quinn-proto/src/tests/util.rs [hunk 2]
```diff
@@ -248,7 +248,9 @@ impl TestEndpoint {
         }
 
         while let Some(x) = self.poll_transmit() {
-            self.outbound.push_back(x);
+            for transmit in split_transmit(x) {
+                self.outbound.push_back(transmit);
+            }
```
Comment (Ralith) [inline quinn-proto/src/tests/util.rs:253]:
```suggestion
            self.outbound.extend(split_transmit(x));
```
and below

**Review by Ralith — COMMENTED** [review]
Maybe this is a good time to make `MAX_DATAGRAMS` a parameter, so this new code isn't dead?

**Comment by Matthias247** [comment]
> Maybe this is a good time to make `MAX_DATAGRAMS` a parameter, so this new code isn't dead?

Indeed. But I think there might be more to discuss around how this would exactly look like, and I decided to try to land this simple change upfront.

**Review by Ralith — APPROVED** [review]

**Pull request merged**